Groups handler and service functions

SamBerry-GDS opened the pull request:
Handlers are moved into folders with code specific to them.

Route into each handler is now index to differentiate from other files.

Services which are shared are now grouped by service under sharedServices rather than all sitting in services.

andylangridge reviewed (changes requested):
Can we change `handlers` to `functions`? And any `index.ts` files to `handler.ts`? 🙏🏻

SamBerry-GDS commented:
> Can we change `handlers` to `functions`? And any `index.ts` files to `handler.ts`? 🙏🏻

I've updated handlers to lambdas and changed the entry point files from index to  handler.
